### PR TITLE
Add logo swiper to homepage.

### PR DESCRIPTION
**Pull request by klslord**
Including animation
屏幕截图 2023-04-18 224718

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -1,3 +1,92 @@
 // The DNS for cdn.jsdelivr.net and fastly.jsdelivr.net have been polluted by the GFW. 
 // For now, gcore.jsdelivr.net is still accessible, but its future availability is uncertain. 
 export const IMG_CDN_URL_PREFIX = 'https://gcore.jsdelivr.net/gh/apache/incubator-kvrocks-website@main/static/img/';
+
+export const userLogoFirstLine = [
```
Comment (PragmaTwice) [inline src/constants.ts:5]: Could the first/second splitting be computed automatically instead of constant definition?
IMHO it is ugly.

Comment (ColinChamber) [reply src/constants.ts:5]: You are right! Here are some issues. The initial consideration was that we wanted to put some well-known companies in the first row. In fact, obtaining the segmentation according to the order of users can also achieve this effect (provided that the maintainer arranges the users in advance according to some rules).

**Review by tisonkun — DISMISSED** [review]
-1

We have a user page to list adopters. No need to push them to the homepage.

**Comment by git-hulk** [comment]
Hi @klslord,

As tison mentioned, we can keep our homepage concise to let new users clearly see what's Kvrocks and its features.

**Comment by ColinChamber** [comment]
Hi tison @tisonkun ,

Indeed, it would be good to include user cases. However, the logo list should also be retained. Here are my thoughts:

1. Many open source projects or organizations display a logo list on their homepage, such as [Nebula](
https://www.nebula-graph.com.cn/)image and [CNCF](https://www.cncf.io/)
image
2. The logo list is constantly scrolling, making the homepage visually more appealing.
3. During our previous discussions in the WeChat group, I asked @git-hulk  and @PragmaTwice  if we still need to display the logo list on the homepage since we have it on other pages. They both suggested that we include it on the homepage as well.

Finally, we will add user cases in the future, similar to the layout on Nebula's homepage. Like this 
image

**Comment by tisonkun** [comment]
@ColinChamber Is a boring game to find examples follow your rule. The following software (not company) doesn't show user logos:

* **Redis** https://redis.io/
* Kafka https://kafka.apache.org/
* Flink https://flink.apache.org/
* Pulsar https://pulsar.apache.org/
* ...

**Comment by ColinChamber** [comment]
Yes, different communities may have different practices. I'm not sure if it's a requirement from Apache for each community to remove a list of logos on the homepage. BTW, you only addressed the first point. Please take look at the other two reasons as I believe visual appeal is also important and a purely static page can be too monotonous.   @tisonkun

**Comment by tisonkun** [comment]
> The logo list is constantly scrolling, making the homepage visually more appealing.

I agree if we have users word, but not a simple logo.

> During our previous discussions in the WeChat group, I asked @git-hulk and @PragmaTwice if we still need to display the logo list on the homepage since we have it on other pages. They both suggested that we include it on the homepage as well.

We don't have a vote for it. Also, suggestions can be thrown and we reach a new consensus if so. So -1 is my vote, it doesn't mean you should follow it.

**Comment by tisonkun** [comment]
> purely static page can be too monotonous

As a homepage for technical software, the landing page can consist of:

* Appealing hero banner
* Key features
* Showcases
* Links to other content

The landing page is to help users understand what this software is and if the software generally fits their usage. Logos help little for these purposes. Leading users with endorsement sentences _may_ help.

**Comment by klslord** [comment]
image
Added user cases, you can add specific content. @git-hulk

File: src/constants.ts (hunk 1 above)
Comment (klslord) [reply src/constants.ts]: solved.

**Comment by ColinChamber** [comment]
I suppose that we still add the logo list at this stage. After kvrocks graduates from the Apache Incubator, I will remember to delete the logo list (return to simplicity).

**Comment by klslord** [comment]
315221098dc7417346f2cf1493c3ec3
Add self-adaption to user cases.

**Comment by PragmaTwice** [comment]
It looks the conflict need to be solved before the next step.

**Comment by klslord** [comment]
Update. @PragmaTwice

**Comment by klslord** [comment]
@PragmaTwice Hulk not find the user comment, so we temporarily remove it and will add it back when available.

**Comment by git-hulk** [comment]
> @PragmaTwice Hulk not find the user comment, so we temporarily remove it and will add it back when available.

Yes, thanks for your efforts. I asked some of the users, they were mainly concerned if it was good to show their words with the company information on the official website. And from my side, it makes sense and we also do not necessarily to show other company logos.

**Comment by ColinChamber** [comment]
Wait for users with a strong recommendation intention.

**Pull request closed without merging**